raspberry/main.py

```diff
@@ -369,48 +374,24 @@ def read_ToF_run_kf_and_LQR():
-                                          in zip(time_cache, throttle_ref_cache, yaw_cache, pitch_cache, roll_cache,
-                                                 tof.altitude_cache(),
-                                                 accelerometer_cache_[:, 0],
-                                                 accelerometer_cache_[:, 1],
-                                                 accelerometer_cache_[:, 2],
-                                                 altitude_reference_cache_mts,
-                                                 radio_data_cache,
-                                                 motor_PWM_cache_[:, 0],
-                                                 motor_PWM_cache_[:, 1],
-                                                 motor_PWM_cache_[:, 2],
-                                                 motor_PWM_cache_[:, 3],
-                                                 KF_data_cache_[:, 0],
-                                                 KF_data_cache_[:, 1],
-                                                 KF_data_cache_[:, 2],
-                                                 KF_data_cache_[:, 3])],
-                                         columns=['timestamp', 'throttle_ref', 'yaw', 'pitch', 'roll',
-                                                  'tof_measurement', 'accX', 'accY', 'accZ',
-                                                  'altitude_ref', 'RC_data', 'mot_FL', 'mot_FR', 'mot_BL', 'mot_BR',
-                                                  'KF_altitude_est', 'KF_velocity_z_est', 'KF_alpha_est', 'KF_beta_est'])
-            data_cache_df.to_csv(
-                f"/home/bzzz/Desktop/logs/data_log_{date_time_now.year}_{date_time_now.month}_{date_time_now.day}_at_{date_time_now.hour}h{date_time_now.minute}m{date_time_now.second}s.csv",
-                index=False,
-                header=True)
-            print("[LOGGER] saving complete")
-            clear_caches()
-            allow_data_logging[0] = False
-
-            # TODO change this to print the dataframe (all rows and columns)
-            if enable_printing_cache_to_screen[0]:
-                print(f"time: {time_cache} \naltitude: {tof.altitude_cache()} \nTref: {throttle_ref_cache} \nyaw: {yaw_cache} \npitch: {pitch_cache} \nroll:{roll_cache} \nacc: {accelerometer_cache_} \nalti_ref_mts{altitude_reference_cache_mts}")
-
-        allow_data_logging[0] = not switch_a_status[0]
+    EVO_filename = datetime.datetime.now().strftime("EVOSENSOR-LOGS-%d-%m-%y--%H-%M.csv")
+    BAR_filename = datetime.datetime.now().strftime("PressureSensor-%d-%m-%y--%H-%M.csv")
+    ANE_filename = datetime.datetime.now().strftime("Anemometer-%d-%m-%y--%H-%M.csv")
+    processor = AverageFilter()  # You need to define this class based on your requirements
+    with EvoSensor(window_length=3,
```

Comment:
Think we should change the layout of this to 

```python
    with (EvoSensor(window_length=3,
                    data_processor=processor,
                    log_file=EVO_filename) as TofSensor, 
          PressureSensor(window_length=100,
                         data_processor=processor,
                         reference_pressure_at_sea_level=102500, 
                         log_file=BAR_filename) as PSensor, 
          Anemometer(window_length=5,
                     data_processor=processor,
                     log_file=ANE_filename) as ASensor):
```

What's your thoughts?

Reply:
Sure